### PR TITLE
ROR-capstone

### PR DESCRIPTION
**Pull request by mbdesigns1989**
Design
You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
Colors.
Typography: font face, size and weight.
Layout: composition and space between elements.
NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.

Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

Interactions
Splash screen

A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
Home page (catagories page)

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
Testing requirements
Create unit and integration tests for all the most important components of your RoR application.
Technical requirements
You should use Postgres as your database.
You should use devise for authentication.
You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
You can use a view template engine of your choice (.erb, .slim, .haml).
The project should be deployed and accessible online.

### CODE REVIEW

File: app/assets/stylesheets/styles.css [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  padding-left: 0.75rem;
+}
+
+.form-style {
+  padding-left: 0;
+  padding-right: 0;
+}
+
+.form-bdy {
+  border-top: 2px solid #ebeaea;
```
Comment (Deepakdanger) [inline app/assets/stylesheets/styles.css:93]: - [ ] I would recommend following the design guideline:
    -  color
Screenshot from 2022-09-06 20-43-44

File: README.md [hunk 2]
```diff
@@ -1 +1,60 @@
...
+This app is about building a mobile web application where you can manage your budget where you have a list of transactions associated with a category, so that you can see how much money you spent and on what.
+
+
+## Live Demo Link
+
+- [Live Link](https://shrouded-castle-53270.herokuapp.com/)
+
+
+## Built With
+- Ruby on Rails
```
Comment (Deepakdanger) [inline README.md:11]: - [ ] Also add the video presentation link in the README file.


  - Record a video presenting the features of the project.
  - The maximum length of the video is 5 minutes.
  - The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
  - The video contains a demo of the project, highlighting all the project features.
  - The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

File: app/views/groups/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            <div id="purchases" class="d-flex flex-column ">
+              <% @purchases.each do |purchase| %>
+                <div class="carrrd ">
+                  <div class="bddy d-flex border  mb-3 pyx bg-body shadow-sm rounded">
+                      <div class="d-flex align-items-center justify-content-evenly fl-1 flex-wrap">
+                        <h3 class="grp-name no-mb mfs-1 fw-bold text-wrap"><%= purchase.name.capitalize %></h3>
+                        <p class="no-mb color-amount fw-bold">$<%= purchase.amount%></p>
+                      </div>
+                  </div>
+                </div>
```
Comment (Deepakdanger) [inline app/views/groups/show.html.erb:32]: - [ ] Here As per design we have to show the date when the transaction is created.

## your design
Screenshot from 2022-09-06 20-08-15

## given design
Screenshot from 2022-09-06 20-07-38

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @mbdesigns1989, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required changes
_Check the comments under the review._


## Optional changes
You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md [hunk 4]
```diff
@@ -1 +1,59 @@
...
+# Budget-app
+This app is about building a mobile web application where you can manage your budget where you have a list of transactions associated with a category, so that you can see how much money you spent and on what.
+
+
+## Live Demo Link
+
+ [live link](https://budget-app-ror.herokuapp.com/) 
+
+## Built With
+- Ruby on Rails
```
Comment (mahbubul14) [inline README.md:10]: - [ ] As the previous reviewer suggested, kindly add the video presentation link in the README file.

instructions

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @mbdesigns1989 ,

Good job so far! :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

# Required Changes ♻️

- [ ] Kindle check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Well implemented the required suggestions.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @mbdesigns1989,

✔️ Good documentation

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**